src/main/java/online/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package online.bookstore.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import online.bookstore.dto.cart.AddCartItemDto;
+import online.bookstore.dto.cart.ShoppingCartDto;
+import online.bookstore.dto.cart.UpdateCartItemQuantityDto;
+import online.bookstore.mapper.CartItemMapper;
+import online.bookstore.mapper.ShoppingCartMapper;
+import online.bookstore.model.CartItem;
+import online.bookstore.model.ShoppingCart;
+import online.bookstore.repository.cart.ShoppingCartRepository;
+import online.bookstore.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addBook(Long id, AddCartItemDto addCartItemDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(id).orElseThrow(()
```

Comment:
Find book by id first
check if book already exists in cart, if yes - increase quantity, if no - add to cart